### PR TITLE
add component disable flag to android build

### PR DESCRIPTION
**Pull request by klatism**
Adding --disable flag to android sdk build script. The flag allows to exclude components from build by concatting -DWITH_XXX cmake flag to the build command. Example : --disable OPENEXR (uppercase).

- [X] I agree to contribute to the project under Apache 2 License.
- [X] To the best of my knowledge, the proposed patch is not based on a code under GPL or another license that is incompatible with OpenCV
- [X] The PR is proposed to the proper branch
- [ ] There is a reference to the original bug report and related work
- [ ] There is accuracy test, performance test and test data in opencv_extra repository, if applicable
      Patch to opencv_extra has the same branch name.
- [ ] The feature is well documented and sample code can be built with the project CMake

### CODE REVIEW

File: platforms/android/build_sdk.py [hunk 1]
```diff
@@ -375,6 +379,7 @@ def get_ndk_dir():
     parser.add_argument('--no_kotlin', action="store_true", help="Disable Kotlin extensions")
     parser.add_argument('--shared', action="store_true", help="Build shared libraries")
     parser.add_argument('--no_media_ndk', action="store_true", help="Do not link Media NDK (required for video I/O support)")
+    parser.add_argument('--disable', metavar='FEATURE', default=[], action='append', help='OpenCV features to disable (add WITH_*=OFF). To disable multiple, specify this flag again, e.g. "--disable tbb --disable openmp"')
```
Comment (opencv-alalek) [inline platforms/android/build_sdk.py:382]: > tbb

Uppercase?

File: platforms/android/build_sdk.py [hunk 2]
```diff
@@ -264,6 +265,9 @@ def build_library(self, abi, do_install, no_media_ndk):
         if no_media_ndk:
             cmake_vars['WITH_ANDROID_MEDIANDK'] = "OFF"
 
+        if len(self.disable) > 0:
+            cmd.extend(["-DWITH_%s=OFF" % f for f in self.disable])
```
Comment (opencv-alalek) [inline platforms/android/build_sdk.py:269]: Use `cmake_vars` instead.

May be even after this line: `cmake_vars.update(abi.cmake_vars)`

**Review by opencv-alalek — COMMENTED** [review]
BTW, this could be handled by using own "config" file for your SDK recipe.

**Comment by klatism** [comment]
thanks for the review! made changes following the comments.

**Comment by klatism** [comment]
@opencv-alalek ready to merge?

**Review by opencv-alalek — APPROVED** [review]
LGTM :+1: Thank you for contribution!

**Pull request merged**